api/src/main/java/org/apache/iceberg/util/DateTimeUtil.java

```diff
@@ -201,7 +229,23 @@ private static int convertMicros(long micros, ChronoUnit granularity) {
     }
   }
 
-  public static long convertNanos(long nanos, ChronoUnit granularity) {
+  public static int nanosToYears(long nanos) {
+    return Math.toIntExact(convertNanos(nanos, ChronoUnit.YEARS));
+  }
+
+  public static int nanosToMonths(long nanos) {
+    return Math.toIntExact(convertNanos(nanos, ChronoUnit.MONTHS));
+  }
+
+  public static int nanosToDays(long nanos) {
+    return Math.toIntExact(convertNanos(nanos, ChronoUnit.DAYS));
+  }
+
+  public static int nanosToHours(long nanos) {
+    return Math.toIntExact(convertNanos(nanos, ChronoUnit.HOURS));
+  }
+
+  private static long convertNanos(long nanos, ChronoUnit granularity) {
```

Comment:
No need to expose `ChronoUnit` from this API. I think this fits with the existing conventions better and is more clear.